x/auth/client/rest/query.go

```diff
@@ -32,15 +33,13 @@ func QueryAccountRequestHandlerFn(
 
 		addr, err := sdk.AccAddressFromBech32(bech32addr)
 		if err != nil {
-			w.WriteHeader(http.StatusBadRequest)
-			w.Write([]byte(err.Error()))
+			utils.WriteErrorResponse(&w, http.StatusInternalServerError, err.Error())
```

Comment:
Lets move this refactor into a seperate PR.

Reply:
requested by @fedekunze - I'd keep it if you don't mind as IMHO it does more good than bad

Reply:
yet I'll refrain in future to tackle technical debt while working on a specific issue :+1: